phoenix-android/src/main/kotlin/fr/acinq/phoenix/android/init/RestoreWalletView.kt

```diff
@@ -41,88 +49,229 @@ import fr.acinq.phoenix.controllers.init.RestoreWallet
+                                Text(stringResource(R.string.restore_error))
+                            }
+                            is RestoreWallet.Model.FilteredWordlist -> {
+                                filteredWords = model.words
+                            }
+
+                            is RestoreWallet.Model.ValidMnemonics -> {
+                                val writingState = vm.writingState
+                                if (writingState is WritingSeedState.Error) {
+                                    Text(
+                                        stringResource(
+                                            id = R.string.autocreate_error,
+                                            writingState.e.localizedMessage
+                                                ?: writingState.e::class.java.simpleName
+                                        )
+                                    )
+                                } else {
+                                    Text(stringResource(R.string.restore_in_progress))
+                                    LaunchedEffect(keyState) {
+                                        vm.writeSeed(
+                                            context,
+                                            vm.selectedWords.toList(),
+                                            false,
+                                            onSeedWritten
+                                        )
+                                    }
                                 }
-                                BorderButton(
-                                    text = R.string.restore_disclaimer_next,
-                                    icon = R.drawable.ic_arrow_next,
-                                    onClick = { showDisclaimer = false },
-                                    enabled = hasCheckedWarning
-                                )
-                            } else {
-                                Text(stringResource(R.string.restore_instructions))
-                                TextInput(
-                                    text = wordsInput,
-                                    onTextChange = { wordsInput = it },
-                                    maxLines = 4,
-                                    modifier = Modifier
-                                        .fillMaxWidth()
-                                        .padding(vertical = 32.dp)
-                                )
-                                BorderButton(
-                                    text = R.string.restore_import_button,
-                                    icon = R.drawable.ic_check_circle,
-                                    onClick = { postIntent(RestoreWallet.Intent.Validate(wordsInput.split(" "))) },
-                                    enabled = wordsInput.isNotBlank()
-                                )
+                            }
+                            else -> {
```

Comment:
Why do we need an `else` case? We already handle all possible cases?

Reply:
Done in e925b518